src/twister2/report/subtest_reporting_plugin.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+import pytest
+from pytest_subtests import SubTestReport
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_report_teststatus(report, config):
+    """ This plugin modifies the one from SubTestReport.
+    We only want to see the subtests' status in the console if verbosity > 1."""
+
+    if report.when != 'call' or not isinstance(report, SubTestReport):
+        return
+
+    if hasattr(report, 'wasxfail'):
+        return None
+
+    # report.location[0] is used when verbosity >=2 by default plugin printing noise
+    # this change the useless path to testcase name
+    loc = list(report.location)
+    loc[0] = f'ztestcase "{report.context.msg}"'
+    report.location = tuple(loc)
+
+    output_formating = {
+        'passed': ['subtests passed', ',', 'SUBPASS'],
+        'skipped': ['subtests skipped', '-', 'SUBSKIP'],
+        'failed': ['subtests failed', 'f', 'SUBFAIL'],
```

Comment:
@fundakol btw the first field tells pytest to which group count the statuses. Here all statuses are counted towards subtests. In the origina implementation only passes are counted towards subtests and other statuses are counted as regular tests
"subtests {output}" everywhere vs "subtests {output}" for passes and just "output" for other statuses in https://github.com/pytest-dev/pytest-subtests/blob/main/pytest_subtests.py#L237
```
    if report.passed:
        return f"subtests {outcome}", ",", "SUBPASS"
    elif report.skipped:
        return outcome, "-", "SUBSKIP"
    elif outcome == "failed":
        return outcome, "u", "SUBFAIL"
```